client/changelist/file_changelist.go

```diff
@@ -94,6 +115,41 @@ func (cl FileChangelist) Close() error {
 	return nil
 }
 
+// NewIterator creates an iterator from FileChangelist
+func (cl FileChangelist) NewIterator() (ChangeIterator, error) {
+	fileInfos, err := getFileNames(cl.dir)
+	if err != nil {
+		return &FileChangeListIterator{}, err
+	}
+	sort.Sort(fileChanges(fileInfos))
+	return &FileChangeListIterator{dirname: cl.dir, collection: fileInfos}, nil
+}
+
+// FileChangeListIterator is a concrete instance of ChangeIterator
+type FileChangeListIterator struct {
+	index      int
+	dirname    string
+	collection []os.FileInfo
+}
+
+// Next returns the next Change in the FileChangeList
+func (m *FileChangeListIterator) Next() (item Change, err error) {
+	defer func() {
+		if e := recover(); e != nil {
+			err = fmt.Errorf("Error in Next(): %v ", e)
+		}
+	}()
+	f := m.collection[m.index]
+	m.index++
+	item, err = unmarshalFile(m.dirname, f)
+	return
+}
+
+// HasNext indicates whether iterator is exhausted
+func (m *FileChangeListIterator) HasNext() bool {
+	return m.index < len(m.collection)
+}
```

Comment:
Won't filesystem errors be returned in the "err" return value? If there's a panic somewhere, that seems like a bug that we should fix instead of arbitrarily adding a recover here.

This code would make sense in a language that uses exceptions for error handling, but this isn't how errors are handled in Go.